src/distributions/other.rs

```diff
@@ -104,6 +121,17 @@ impl Distribution<u8> for Alphanumeric {
     }
 }
 
+#[cfg(feature = "alloc")]
+impl DistString for Alphanumeric {
+    fn append_string<R: Rng + ?Sized>(&self, rng: &mut R, string: &mut String, len: usize) {
+        unsafe {
+            let v = string.as_mut_vec();
+            v.reserve(v.len() + len);
```

Comment:
Parameter is incorrect. But `reserve` is not necessary here; `extend` will reserve the correct amount based on the upper bound of the `Take` iterator.